base/floatfuncs.jl

```diff
@@ -105,24 +105,30 @@ end
 
 function _signif_og(x, digits, base)
     if base == 10
-        oftype(x, 10. ^ floor(log10(abs(x)) - digits + 1.))
+        e = floor(log10(abs(x)) - digits + 1.)
+        og = oftype(x, exp10(abs(e)))
     elseif base == 2
-        oftype(x, 2. ^ floor(log2(abs(x)) - digits + 1.))
+        e = exponent(abs(x)) - digits + 1.
+        og = oftype(x, exp2(abs(e)))
```

Comment:
No, `exp2` converts integer arguments to floats, and all float arguments (even if integer valued) should be roughly the same speed.

Let's go with what we have now: there is always room for more speed hacks later on, it's probably more important to have it correct for the time being.